### PR TITLE
PWM driver for xmc4xxx using ccu4 and ccu8 module

### PR DESCRIPTION
**Pull request by talih0**
Adds drivers for pwm on xmc4xxx using Capture Compare Unit 4 (CCU4)
module. There are four CCU4 modules with each one having four channels
(the channels are called slices in infineon documentation).
Thus it's possible to have up to 16 pwm output signals. The output of
each channel can only be connected to a specific port/pin. The possible
connections and gpio configurations are defined using pinctrl.

### CODE REVIEW

**Review by ifyall — CHANGES_REQUESTED** [review]

File: drivers/pwm/pwm_xmc4xxx_ccu4.c [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	/* enables the CCU4 clock and ungates clock to CCU4x */
+	XMC_CCU4_EnableModule(config->ccu4);
+	XMC_CCU4_StartPrescaler(config->ccu4);
+
+	for (int i = 0; i < NUM_SLICES; i++) {
+		XMC_CCU4_SLICE_t *slice;
+		XMC_CCU4_SLICE_COMPARE_CONFIG_t slice_conf = {.prescaler_initval =
+								      config->prescaler[i]};
+
+		slice = (XMC_CCU4_SLICE_t *)((uint32_t)config->ccu4 + (i + 1) * 0x100);
```
[the remark this answers is not in the document]
Comment (talih0) [reply drivers/pwm/pwm_xmc4xxx_ccu4.c]: thanks @ifyall, I updated the PR

**Comment by talih0** [comment]
I've updated PR to add PWM via CCU8, which can  generate complementary high-side/low-side signals for each output channel. A variable dead time can be added during the off to on transitions to make sure that the high-side/low-side signals are not on at the same time.

**Comment by talih0** [comment]
Hi @ifyall, @npal-cy, could you have another look at this PR when you have a moment pls?
It's a dependency for this PR: https://github.com/zephyrproject-rtos/zephyr/pull/59138

**Review by ifyall — DISMISSED** [review]

**Comment by talih0** [comment]
@ifyall, I rebased to fix merge conflicts. Could you re-ack pls

**Review by ifyall — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**